src/libsyntax/unicode/tables.rs

```diff
@@ -0,0 +1,595 @@
+// Copyright 2012-2014 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// NOTE: The following code was generated by "src/etc/unicode.py", do not edit directly
```

Comment:
I could pull `librustc_unicode` into syntex but for the fact that it includes the `char` lang item. What if we split it into two crates? One has the tables and the unicode algorithms, and the other has the `char` lang item?